vignettes/epi_df.Rmd

```diff
@@ -209,35 +209,117 @@ Epidata API](https://cmu-delphi.github.io/delphi-epidata/api/flusurv.html)
+uses either [NCHS](https://www.cdc.gov/nchs/nvss/bridged_race.htm) or
+[Census](https://www.census.gov/programs-surveys/popest/technical-documentation/research/evaluation-estimates/2020-evaluation-estimates/2010s-county-detail.html)
+populations for `time_value`s before 2020 included in reports from 2020 onward,
+but in this case, the two sources agreed exactly. FluSurv-NET also directly
+reports an overall rate, so we can check our work.
 ```{r}
-group_cols <- key_colnames(exclude = "age_group")
-flu_data %>%
-  sum_groups_epi_df("rate", group_cols = group_cols)
+group_cols <- key_colnames(flu_data, exclude = "age_group")
+rate_overall_recalc_edf <-
+  flu_data %>%
+  inner_join(
+    # Population estimates for FluSurv-NET-covered part of CA on 2017-07-01 and
+    # 2018-07-01, extracted and aggregated from "vintage 2020" estimates (actually
+    # released by Census in June 2021 and by NCHS in September 2021), which is the
+    # last available reporting found with population estimates for 2017 and 2018:
+    tribble(
+      ~geo_value, ~age_group,  ~time_value,           ~pop,
+      "CA",       "0--4 yr",   as.Date("2017-07-01"), 203813,
+      "CA",       "5--17 yr",  as.Date("2017-07-01"), 521827,
+      "CA",       "18--49 yr", as.Date("2017-07-01"), 1722399,
+      "CA",       "50--64 yr", as.Date("2017-07-01"), 700090,
+      "CA",       ">= 65 yr",  as.Date("2017-07-01"), 534789,
+      "CA",       "0--4 yr",   as.Date("2018-07-01"), 201265,
+      "CA",       "5--17 yr",  as.Date("2018-07-01"), 520077,
+      "CA",       "18--49 yr", as.Date("2018-07-01"), 1725382,
+      "CA",       "50--64 yr", as.Date("2018-07-01"), 699145,
+      "CA",       ">= 65 yr",  as.Date("2018-07-01"), 551243,
+    ),
+    # Simple population interpolation/extrapolation scheme: last observation
+    # carried forward. Use the estimated population on 2017-07-01 for
+    # time_values 2017-07-01 through 2018-06-30, and the estimated population on
+    # 2018-07-01 for all subsequent time_values:
+    join_by(geo_value, closest(y$time_value <= x$time_value), age_group),
+    suffix = c("", "_for_pop"),
+    relationship = "many-to-one", unmatched = "error"
+  ) %>%
+  select(-time_value_for_pop) %>%
+  group_by(geo_value, time_value) %>%
+  mutate(count = rate * pop / 100e3) %>%
+  ungroup() %>%
+  sum_groups_epi_df(c("count", "pop"), group_cols = group_cols) %>%
+  mutate(rate_overall_recalc = count / pop * 100e3) %>%
+  # match rounding of original data:
+  mutate(rate_overall_recalc = round(rate_overall_recalc, 1)) %>%
+  # compare to published overall rates:
+  inner_join(
+    flu_data_api %>%
+      select(geo_value = location, time_value = epiweek, rate_overall),
+    by = c("geo_value", "time_value"),
+    relationship = "one-to-one", unmatched = "error"
+  )
+# What's our maximum error vs. the official overall estimates?
+max(abs(rate_overall_recalc_edf$rate_overall - rate_overall_recalc_edf$rate_overall_recalc))
```

Comment:
suggestion: also separate this out into another chunk.

Reply:
I also addressed this in https://github.com/cmu-delphi/epiprocess/pull/591/commits/b569131305741c12bad05397bd05524852814b21, but if you decide not to keep that commit, would need to re-separate some of this code.